### ISSUE TITLE
Namespace does not reflect in the UI to install service mesh

### ISSUE THREAD

**Issue by bluedog13** [meshery/meshery#6995]
#### Current Behavior
Not able to install service mesh in the desired namespace using the UI.  The required namespaces does not show in the UI to select. ex: linkerd

#### Desired Behavior
Option to select the desired namespace in the UI

#### Screenshots/Logs
I have the below namespaces created in my k8s cluster

```
kubectl get namespaces

--
default           Active   41h
kube-node-lease   Active   41h
kube-public       Active   41h
kube-system       Active   41h
linkerd           Active   15m
meshery           Active   11m
```

To install LinkerD service mesh in my cluster, I am not able to select the "linkerd" namespace in the UI for installation. All I see is the default namespace.  It also won't accept when I type the namespace linkerd

![Screenshot 2023-02-03 at 11 36 07 AM](https://user-images.githubusercontent.com/11997472/216669901-3e8ca182-80e4-42e7-af05-307705c51207.png)

#### Environment

- Browser: Chrome 
- Host OS: Mac 13.2 (22D49)
- Client	v0.6.49	460e9bd43
- Server	v0.6.47	4c357f30
- Platform: Kubernetes
---

Few comments on the currentUI documentation
1. The UI does not match any of the screenshots on the website or examples.  As an example, the "Dashboard" selection exists in my UI and not on the website examples.  Perhaps this is due to the version mismatch, but it makes adoption and usability experience harder than what it's supposed to be. 

2. Larger screen shots by default instead of clicking on the smaller ones  to enlarge will also help and save time for the end users. 

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

**Comment by Revolyssup** [comment]
@bluedog13 Can you make sure that you are connected to meshery-broker? The data from your Kubernetes cluster is streamed back via meshery-broker which will have this information about available namespaces as well. You can manually enter "linkerd", or go to the settings page and click on your connected Kubernetes context to check if MeshSync is active.
<img width="1015" alt="Screenshot 2023-02-03 at 11 25 27 PM" src="https://user-images.githubusercontent.com/43276904/216673520-50d223d4-e4b4-4f2f-a3dd-ca6084f1ac95.png">

**Comment by bluedog13** [comment]
When I enter linkerd manually and tab out - it gets wiped out and goes back to Select.... in the drop down box. 

I am connected to the meshery-broker.   

![Screenshot 2023-02-03 at 12 04 26 PM](https://user-images.githubusercontent.com/11997472/216675253-44c3b8b2-6f09-4dee-8410-7317cecd11dc.png)

**Comment by Revolyssup** [comment]
> When I enter linkerd manually and tab out - it gets wiped out and goes back to Select.... in the drop down box.
> 
> I am connected to the meshery-broker.
> 
> ![Screenshot 2023-02-03 at 12 04 26 PM](https://user-images.githubusercontent.com/11997472/216675253-44c3b8b2-6f09-4dee-8410-7317cecd11dc.png)

For manual entry - You'll have to press return after entering the namespace.
It seems like your NATS is connected just fine. Can you check Meshery logs to see if there are any errors related to not being able to connect to broker?

**Comment by bluedog13** [comment]
Pressing "Enter" also does not do anything.  The cursor just remains in the dropdown box.   If I then click on the "manage service mesh lifecycle" box,  the Namespace dropdown goes back to Select....

Where can I find mesher logs and what exactly in the logs am I looking for?

**Comment by bluedog13** [comment]
When I use  the meshery docker extension and run mesherysystem dashboard - the UI opens on port localhost:9081

If I do the same without using docker extension,  the UI opens on port localhost: 300XXX but this page shows connection issue.  The dashboard is stlil available on 9081. 

Where is the port 30xxx setup and is this the reason for the namespace now showing up problem?

I just used docker extension to log into meshery and I am able to see namespaces now. But the same kubernetes cluster shows up with a different name (in-cluster vs docker-desktop as above)

![Screenshot 2023-02-03 at 12 20 21 PM](https://user-images.githubusercontent.com/11997472/216678044-206fef4d-dc63-45fb-b9b9-c115891c93ff.png)

**Comment by bluedog13** [comment]
Below is one of the ways to make it work ( which makes me believe this is a meshery UI application bug )

Delete the context from the top right menu by clicking on the "x" mark next to the context.
![Screenshot 2023-02-06 at 3 46 39 PM](https://user-images.githubusercontent.com/11997472/217095174-fe257114-78a6-4562-b6a3-cb399814f89b.png)


After this, the existing context gets added again automatically and will reflect all the namespaces in the k8s cluster.

**Comment by leecalcote** [comment]
@MUzairS15 @Abhishek-kumar09, please review.

**Comment by MUzairS15** [comment]
On it

**Comment by Revolyssup** [comment]
> When I use the meshery docker extension and run mesherysystem dashboard - the UI opens on port localhost:9081
> 
> If I do the same without using docker extension, the UI opens on port localhost: 300XXX but this page shows connection issue. The dashboard is stlil available on 9081.
> 
> Where is the port 30xxx setup and is this the reason for the namespace not showing up problem?
> 
> I just used docker extension to log into meshery and I am able to see namespaces now. But the same kubernetes cluster shows up with a different name (in-cluster vs docker-desktop as above)
> 
> ![Screenshot 2023-02-03 at 12 20 21 PM](https://user-images.githubusercontent.com/11997472/216678044-206fef4d-dc63-45fb-b9b9-c115891c93ff.png)

the 3xxxx port is nodeport and you're unable to see the namespaces due to network connectivity. `meshery-broker` service of type `LoadBalancer` was not allocated an ExternalIP.

**Comment by MUzairS15** [comment]
Yes, @Revolyssup is right, I was able to get all namespaces listed once the connection was established with Broker.
You need to look for log that says “Connected to broker at endpoint <endpoint >”.

Regarding, different cluster name, the name you see on the kubernetes chip is context name, which is entirely a user defined construct, to bind user and cluster and k8s has no such meaning to it.

When you deploy meshery outside Kubernetes, you provide your Kubeconfig, to connect to cluster and from their meshery is able to get the name.

when meshery is deployed on a cluster, it is automatically connected to that same cluster and since no config was provided, a default name is given.

**Comment by bluedog13** [comment]
Couple things in my case to consider  if they were not clear earlier

1.  Meshery is running inside the local k8s cluster and the cluster is started using docker-desktop

2. Even without specifying the k8s config file, when I apply the hack above (as indicated in my post), I am able to see all the namespaces (newly created) that are not listed otherwise in the meshery UI. 

3. I am no longer using the meshery docker extension and still with the above hack, the namespaces are listed correctly every time I use the hack. Otherwise, the newly created namespaces don't show up.  I've seen the problem happen even when using the docker extension and hence gave up using docker extension. 

So this makes me think - it's not related to any config file, meshery broker and such.  If this is the case, the manual hack should also not work.  It's related to the UI that is not able to read the latest state in the k8s cluster.
